receiver/awscontainerinsightreceiver/internal/neuron/pod_attribute_decorator.go

```diff
@@ -0,0 +1,131 @@
+package neuron
+
+import (
+	"context"
+	"strconv"
+
+	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+const (
+	neuronHardwareInfoKey       = "neuron_hardware"
+	neuronCorePerDeviceKey      = "neuroncore_per_device_count"
+	neuronCoreAttributeKey      = "neuroncore"
+	neuronDeviceAttributeKey    = "neuron_device_index"
+	neuronCoreResourceName      = "aws.amazon.com/neuroncore"
+	neuronDeviceResourceName    = "aws.amazon.com/neurondevice"
+	neuronDeviceResourceNameAlt = "aws.amazon.com/neuron"
+)
+
+type PodAttributesDecoratorConsumer struct {
+	NextConsumer      consumer.Metrics
+	PodResourcesStore *stores.PodResourcesStore // replace with podResourcesApi
+	Logger            *zap.Logger
+}
+
+func (pdc *PodAttributesDecoratorConsumer) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{
+		MutatesData: true,
+	}
+}
+
+func (pdc *PodAttributesDecoratorConsumer) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) error {
+	md = pdc.neuronMetricsProcess(md)
+	return pdc.NextConsumer.ConsumeMetrics(ctx, md)
+}
+
+func (pdc *PodAttributesDecoratorConsumer) neuronMetricsProcess(md pmetric.Metrics) pmetric.Metrics {
```

Comment:
No need to return `pmetric.Metrics` since you modify the original in place (as you should).